### PR TITLE
Log interactions to MySQL

### PR DESCRIPTION
**Pull request by mruwnik**
* Adds MySQL support
* Logs interactions to the database

This will require the db env variables to be set in GH actions

### CODE REVIEW

File: api/src/stampy_chat/logging.py [hunk 1]
```diff
@@ -1,28 +1,61 @@
...
         return self.isEnabledFor(DEBUG)
 
-    def query(self, query):
+    def interaction(self, query, response, history, prompt, blocks):
+        prompt = [i for i in prompt if i.get('role') == 'system']
+        prompt = prompt[0].get('content') if prompt else None
+
+        self.item_adder.add(
+            Interaction(
+                # session_id=session_id,
```
Comment (mruwnik) [inline api/src/stampy_chat/logging.py:24]: this isn't handled by the frontend yet, so isn't passed in

File: local_db.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
```
Comment (mruwnik) [inline local_db.sh:1]: a helper script to set up the database locally as a docker container. This is the same as for the ARD, but the container has a different name

File: api/src/stampy_chat/db/models.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+class Base(DeclarativeBase):
+    pass
+
+
+class Interaction(Base):
+    __tablename__ = "interactions"
+
+    id: Mapped[int] = mapped_column("id", primary_key=True)
+    session_id: Mapped[str] = mapped_column(UUID(), default=uuid.uuid4)
```
[the remark this answers is not in the document]
Comment (FraserLee) [reply api/src/stampy_chat/db/models.py:45]: good choice

**Comment by FraserLee** [comment]
Good work setting the PR branch of dependent PRs as the target of the next in the chain. Github has awful support for dependent PRs, but I personally really like this workflow \:)

File: api/src/stampy_chat/chat.py [hunk 4]
```diff
@@ -238,14 +228,14 @@ def talk_to_robot_internal(index, query: str, mode: str, history: List[Dict[str,
         yield {'state': 'error', 'error': str(e)}
 
 # convert talk_to_robot_internal from dict generator into json generator
-def talk_to_robot(index, query: str, mode: str, history: List[Dict[str, str]], k: int = STANDARD_K, log: Callable = logger.info):
```
Comment (FraserLee) [inline api/src/stampy_chat/chat.py:241]: Ah I'm seeing you caught this here. Might be better to do these changes on #86 and then merge from `log-queries` up into `mysql-logs`.

File: api/src/stampy_chat/db/models.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    interaction_no: Mapped[int] = mapped_column(Integer)
```
[the remark this answers is not in the document]
Comment (FraserLee) [reply api/src/stampy_chat/db/models.py:46]: Can you turn this into a comment in the file?

File: api/src/stampy_chat/get_blocks.py
Comment (FraserLee) [inline api/src/stampy_chat/get_blocks.py:1]: again, if possible, unrelated changes should be their own PRs.

File: readme.md
Comment (FraserLee) [inline readme.md:1]: Needs *a lot* more documentation on how to setup and test in a dev environment.

**Review by FraserLee — CHANGES_REQUESTED** [review]

**Comment by mruwnik** [comment]
moved to other PRs, so closing this one

**Pull request closed without merging**